src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java

```diff
@@ -167,6 +164,106 @@ public ResponseEntity<byte[]> getTemplateFile(@PathVariable ProgrammingLanguage
     }
 
+    /**
+     * GET /files/aeolus/templates/:language/:projectType : Get the aeolus template file with the given filename<br/>
+     * GET /files/aeolus/templates/:language : Get the aeolus template file with the given filename
+     * <p>
+     * The windfile contains the default build plan configuration for new programming exercises.
+     *
+     * @param language       The programming language for which the aeolus template file should be returned
+     * @param projectType    The project type for which the template file should be returned. If omitted, a default depending on the language will be used.
+     * @param staticAnalysis Whether the static analysis template should be used
+     * @param sequentialRuns Whether the sequential runs template should be used
+     * @param testCoverage   Whether the test coverage template should be used
+     * @return The requested file, or 404 if the file doesn't exist
+     */
+    @GetMapping({ "files/aeolus/templates/{language}/{projectType}/{staticAnalysis}/{sequentialRuns}/{testCoverage}",
+            "files/aeolus/templates/{language}/{staticAnalysis}/{sequentialRuns}/{testCoverage}" })
+    @EnforceAtLeastEditor
+    public ResponseEntity<String> getAeolusTemplate(@PathVariable ProgrammingLanguage language, @PathVariable Optional<ProjectType> projectType,
+            @PathVariable Optional<Boolean> staticAnalysis, @PathVariable Optional<Boolean> sequentialRuns, @PathVariable Optional<Boolean> testCoverage) {
+        log.debug("REST request to get aeolus template for programming language {} and project type {}, static Analysis: {}, sequential Runs {}, testCoverage: {}", language,
+                projectType, staticAnalysis, sequentialRuns, testCoverage);
+
+        String languagePrefix = language.name().toLowerCase();
+        String projectTypePrefix = projectType.map(type -> type.name().toLowerCase()).orElse("");
+
+        return getAeolusTemplateFileContentWithResponse(languagePrefix, projectTypePrefix, staticAnalysis.orElse(false), sequentialRuns.orElse(false), testCoverage.orElse(false));
+    }
+
+    /**
+     * Returns the file content of the template file for the given language and project type as JSON
+     *
+     * @param languagePrefix    The programming language for which the template file should be returned
+     * @param projectTypePrefix The project type for which the template file should be returned. If omitted, a default depending on the language will be used.
+     * @param staticAnalysis    Whether the static analysis template should be used
+     * @param sequentialRuns    Whether the sequential runs template should be used
+     * @param testCoverage      Whether the test coverage template should be used
+     * @return The requested file, or 404 if the file doesn't exist
+     */
+    private ResponseEntity<String> getAeolusTemplateFileContentWithResponse(String languagePrefix, String projectTypePrefix, Boolean staticAnalysis, Boolean sequentialRuns,
```

Comment:
```suggestion
    private String getAeolusTemplateFileContentWithResponse(String languagePrefix, String projectTypePrefix, boolean staticAnalysis, boolean sequentialRuns,
```
When moving to a separate service this should return a base type and wrapped into the response here in the RESTcontroller.

I would also prefer primitive `boolean`s since the third option with `null` does not really make sense here.